File: Data Structures & Algorithms/sliding-window-maximum/submission-1.java
class Solution {
    // public int[] maxSlidingWindow(int[] nums, int k) {
    //     int l=0;
    //     List<Integer> res = new ArrayList<>();
    //     for(int r=0; r<nums.length; r++){
    //         if(r-l+1 == k){
    //             res.add(findMax(nums,l,r));
    //             l++;
    //         }
    //     }
    //     int[] ans = new int[res.size()];
    //     for(int i=0; i<res.size();i++){
    //         ans[i] = res.get(i);
    //     }
    //     return ans;
    // }
    // int findMax(int[] arr, int s, int e){
    //     int max = Integer.MIN_VALUE;
    //     for(int i=s; i<=e; i++){
    //         max = Math.max(max,arr[i]);
    //     }
    //     return max;
    // }
    
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n = nums.length;
        int[] output = new int[n - k + 1];

        for (int i = 0; i <= n - k; i++) {
            int maxi = nums[i];
            for (int j = i; j < i + k; j++) {
                maxi = Math.max(maxi, nums[j]);
            }
            output[i] = maxi;
        }

        return output;
    }

}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [-1] * len(nums) 
        res = 0
        for i in range(len(nums)):
            res = max(res, self.dfs(nums, i, dp))
        return res
    
    def dfs(self, nums, i, dp):
        if i >= len(nums):
            return 0
        if dp[i] != -1:
            return dp[i]
        res = 1
        for j in range(i+1, len(nums)):
            curr_len= 1
            if nums[j] > nums[i]:
                curr_len += self.dfs(nums, j, dp)
                res = max(res, curr_len)
        dp[i] = res
        return res


            